activation/activation_test.go

```diff
@@ -468,16 +468,29 @@ func TestBuilder_RestartSmeshing(t *testing.T) {
 		layerClockMock, &mockSyncer{}, logtest.New(t).WithName("atxBuilder"))
 	builder.initialPost = initialPost
 
-	for i := 0; i < 100; i++ {
-		require.NoError(t, builder.StartSmeshing(types.Address{}, atypes.PostSetupOpts{}))
-		// NOTE(dshulyak) this is a poor way to test that smeshing started and didn't exit immediately,
-		// but proper test requires adding quite a lot of additional mocking and general refactoring.
-		time.Sleep(400 * time.Microsecond)
-		require.Truef(t, builder.Smeshing(), "failed on execution %d", i)
-		require.NoError(t, builder.StopSmeshing(true))
-		time.Sleep(400 * time.Microsecond)
-		require.Falsef(t, builder.Smeshing(), "failed on execution %d", i)
-	}
+	t.Run("Single threaded", func(t *testing.T) {
+		for i := 0; i < 100; i++ {
+			require.NoError(t, builder.StartSmeshing(types.Address{}, atypes.PostSetupOpts{}))
+			require.Never(t, builder.Finished, 400*time.Microsecond, 50*time.Microsecond, "failed on execution %d", i)
```

Comment:
It looks a bit convoluted, but this way you don't need both `builder.Smeshing` and `builder.Finished`:

```suggestion
			require.Never(t, func() bool { return !builder.Smeshing() }, 400*time.Microsecond, 50*time.Microsecond, "failed on execution %d", i)
```

Reply:
I added `Finished()` to make tests more readable :D